pinot-core/src/main/java/org/apache/pinot/core/operator/timeseries/TimeSeriesAggregationOperator.java

```diff
@@ -129,6 +140,12 @@ protected TimeSeriesResultsBlock getNextBlock() {
           throw new IllegalStateException(
               "Don't yet support value expression of type: " + valueExpressionBlockValSet.getValueType());
       }
+      Preconditions.checkState(seriesBuilderMap.size() * (long) _timeBuckets.getNumBuckets() <= _maxDataPointsLimit,
+          "Exceeded max data point limit per server. Limit: %s. Data points in current segment so far: %s",
+          _maxDataPointsLimit, seriesBuilderMap.size() * _timeBuckets.getNumBuckets());
+      Preconditions.checkState(seriesBuilderMap.size() <= _maxSeriesLimit,
+          "Exceeded max unique series limit per server. Limit: %s. Series in current segment so far: %s",
+          _maxSeriesLimit, seriesBuilderMap.size());
     }
```

Comment:
do we intend to have a partial result when numSeries > maxSeries something similar to numGroupsLimitReached?

Reply:
Nope our current plan is to fail the query altogether. We want to stay away from partial results since showing warnings to the different user interfaces ends up being quite complex and has a bunch of pitfalls.